mitodl/micromasters
# Updating MicromastersCertificateModel to relate to User and Course

#### What are the relevant tickets?
Related to #3870

#### What's this PR do?
Updating `MicromastersCertificateModel` part 1.
Adding `user` and `course` fields, making `final_grade` nullable.
Populating the new fields.
Changing the logic involving certificates.

#### How should this be manually tested?
To create a certificate for a user:
Run:
`alter_data set_to_passed --username "username" --course-title 'Digital Learning 200'`
Then for your `user`:
```
from courses.models import *
from grades.models import *
from grades.factories import *
course = Course.objects.filter(title='Digital Learning 200')[0]
course_run = course.courserun_set.all().first()
final_grade = FinalGradeFactory.create(user=user, course_run=course_run, passed=True, status='complete', grade=0.8)
CourseRunGradingStatus.objects.create(course_run=course_run, status='complete')
cert = MicromastersCourseCertificate.objects.create(final_grade=final_grade, user=user, course=course)
```
If you are creating second certificate make sure that you pick a different course run for that course.

If you want to see the certificate in the dashboard, also run this:
```
from cms.factories import *
CourseCertificateSignatoriesFactory.create(course=course)
```
